frontend/catalyst/jit.py

```diff
@@ -576,11 +578,40 @@ def capture(self, args, **kwargs):
         verify_static_argnums(args, self.compile_options.static_argnums)
         static_argnums = self.compile_options.static_argnums
         abstracted_axes = self.compile_options.abstracted_axes
+        experimental_capture = self.compile_options.experimental_capture
 
         dynamic_args = filter_static_args(args, static_argnums)
         dynamic_sig = get_abstract_signature(dynamic_args)
         full_sig = merge_static_args(dynamic_sig, args, static_argnums)
 
+        def fn_with_transform_named_sequence(*args, **kwargs):
+            """
+            This function behaves exactly like the user function being jitted,
+            taking in the same arguments and producing the same results, except
+            it injects a transform_named_sequence jax primitive at the beginning
+            of the jaxpr when being traced.
+
+            Note that we do not overwrite self.original_function and self.user_function;
+            this fn_with_transform_named_sequence is ONLY used here to produce tracing
+            results with a transform_named_sequence primitive at the beginning of the
+            jaxpr. It is never executed or used anywhere, except being traced here.
+            """
+            _inject_transform_named_sequence()
+            return self.user_function(*args, **kwargs)
+
+        if experimental_capture:
+
+            jaxpr, out_type, treedef = trace_to_jaxpr(
+                fn_with_transform_named_sequence,
+                static_argnums,
+                abstracted_axes,
+                experimental_capture,
+                full_sig,
+                kwargs,
+            )
+
+            return jaxpr, out_type, treedef, dynamic_sig
```

Comment:
This block surprised me a bit because it both checks the condition and passes it on to `trace_to_jaxpr`, but I guess the purpose is to avoid the patching. Possible alternative, but this is optional:
```py
        with Patcher(
            (qml.QNode, "__call__", closure if experimental_capture else qml.QNode.__call__),
        ):
```